src/PlanetData.js

```diff
@@ -25,49 +44,81 @@ PlanetData.prototype = {
-      for (var y = 0; y < this.height; y++) {
-        buffer[x].push(noise.fbm2(x * 0.005, y * 0.005));
+      rivers.push([]);
+      for (var y = 0; y < this.mapData.canvas.height; y++) {
+        terh = this.heightmap(x * this.skew, y);
+        rv = 1 - noise.worley2(x * 25 * this.rfrequency + terh * 3, y * 25 * this.rfrequency + terh * 3);//curve the river based on height
+        rivers[x].push(rv);
+        buffer[x].push(terh - rv * 0.8 * this.erodibility * (1 - terh) * this.moisture);//decrease height around rivers
+        if (terh > peak) {
+          peak = terh;
+        } else if (terh < low) {
+          low = terh;
+        }
+        
       }
     }
-
+    //TODO: add texture to water
     //Here we will take the height data and convert it to colors
     var alpha, red, green, blue, h, c, i;
-    for (var x = 0; x < this.width - 1; x++) {
-      for (var y = 0; y < this.height - 1; y++) {
+    var waterLevel = utils.lerp(low, peak, utils.smoothstep(low, peak, this.moisture) * 0.3);
+    for (var x = 1; x < buffer.length - 1; x++) {
+      for (var y = 1; y < buffer[x].length - 1; y++) {
         h = buffer[x][y];
-        alpha = 255;
+        rv = rivers[x][y];
+        var w = 0;
+
+        //this block determines where the rivers are
+        /*todo: use an analytic derivitive to compute w rather than a logic block*/
+        if (x > 1 && y > 1) {
+          if (((rv >= rivers[x + 1][y] && rv >= rivers[x - 1][y])||(rv >= rivers[x][y + 1] && rv >= rivers[x][y - 1]))&&(rv > (0.7 - this.moisture))) {
+            w = peak - h;
+          }
+        }
+        //determines what the color of the river will be so that they can fade in towards sea level
+        var rl = utils.mix(this.landHue, this.waterHue, utils.smoothstep(0, peak - waterLevel, w * this.moisture));
         //This takes the height and maps the land or water color to it
-        //the smoothstep means that below 0.4 is solid land
-        //then between 0.4-0.7 there is a smooth gradient
-        //and after it is purely water
-        c = utils.mix(this.landHue, this.waterHue, utils.smoothstep(0.4, 0.7, h));
+        //the smoothstep means that below 0.3 is water
+        //then between 0.3-0.6 there is a smooth gradient
+        //and after it is purely land
+        c = utils.mix(this.waterHue, rl, utils.smoothstep(waterLevel, waterLevel + 0.2 * this.moisture, h));
+        
         //calculate the normal at a given pixel
         //this allows us to cheaply shadow the terrain
         var d = new Phaser.Point(h - buffer[x + 1][y], h - buffer[x][y + 1]);
         d = d.normalize();
         //map our normal to the range 0-1
         d.x = 0.5 * (d.x + 1);
         //this removes the normal from the water area so we only see it on land
-        d.x = utils.lerp(d.x, 1, utils.smoothstep(0.5, 0.6, h));
+        d.x = utils.lerp(1, d.x, utils.smoothstep(waterLevel + 0.1 * this.moisture, waterLevel + 0.2 * this.moisture, h - w * 0.5 * this.moisture));

```

Comment:
Let's calculate 
`waterLevel + 0.1 * this.moisture, waterLevel + 0.2 * this.moisture` and 
` h - w * 0.5 * this.moisture` into named variables above for clarity and shorter lines :)